lib/SparqlXmlParser.ts

```diff
@@ -29,79 +28,112 @@ export class SparqlXmlParser {
-      .pipe(new SparqlXmlBindingsTransformer(this));
-
-    // Propagate errors
-    sparqlResponseStream.on('error', (error) => resultStream.emit('error', error));
-
-    return resultStream;
-  }
-
-  /**
-   * Convert a SPARQL XML result binding to a bindings object.
-   * @param rawBindings A SPARQL XML result binding.
-   * @return {IBindings} A bindings object.
-   */
-  public parseXmlBindings(rawBindings: any): IBindings {
-    const bindings: IBindings = {};
-    if (rawBindings.children) {
-      const bindingsArray = Array.isArray(rawBindings.children.binding)
-        ? rawBindings.children.binding : [rawBindings.children.binding];
-      for (const binding of bindingsArray) {
-        if (binding.attribs && binding.children) {
-          const key = binding.attribs.name;
-          let term: RDF.Term = null;
-          if (binding.children.bnode) {
-            term = this.dataFactory.blankNode(binding.children.bnode.value);
-          } else if (binding.children.literal) {
-            const value = binding.children.literal.value || "";
-            const attribs = binding.children.literal.attribs;
-            if (attribs && attribs['xml:lang']) {
-              term = this.dataFactory.literal(value, attribs['xml:lang']);
-            } else if (attribs && attribs.datatype) {
-              term = this.dataFactory.literal(value, this.dataFactory.namedNode(attribs.datatype));
-            } else {
-              term = this.dataFactory.literal(value);
-            }
-          } else {
-            term = this.dataFactory.namedNode(binding.children.uri.value);
+        .pipe(new Transform({
+          objectMode: true,
+          transform(chunk: any, encoding: string, callback: (error?: Error | null, data?: any) => void) {
+            parser.write(chunk);
+            callback();
           }
-          bindings[this.prefixVariableQuestionMark ? ('?' + key) : key] = term;
-        }
-      }
-    }
-    return bindings;
+        }));
+    return resultStream;
   }
 
   /**
    * Convert a SPARQL XML boolean response stream to a promise resolving to a boolean.
    * This will reject if the given response was not a valid boolean response.
    * @param {NodeJS.ReadableStream} sparqlResponseStream A SPARQL XML response stream.
-   * @return {NodeJS.ReadableStream} A stream of bindings.
+   * @return {Promise<boolean>} The response boolean.
    */
   public parseXmlBooleanStream(sparqlResponseStream: NodeJS.ReadableStream): Promise<boolean> {
     return new Promise((resolve, reject) => {
-      sparqlResponseStream.on('error', reject);
+      const parser = new SaxesParser();
+      const stack: string[] = [];
+      parser.on("error", reject);
+      parser.on("opentag", tag => {
+        stack.push(tag.name);
+      })
+      parser.on("closetag", _ => {
+        stack.pop();
+      })
+      parser.on("text", text => {
+        if(this.stackEquals(stack, ['sparql', 'boolean'])) {
+          resolve(text === 'true');
```

Comment:
Sorry for that. I have just added a test `should reject a results payload` (c.f. my comment on the code review interface) and the current code seems to reject the payload properly because of the `reject(` function call inside of the `end` event handler.